daemon/graphdriver/aufs/aufs.go

```diff
@@ -350,6 +350,10 @@ func (a *Driver) Put(id string) error {
 // Diff produces an archive of the changes between the specified
 // layer and its parent layer which may be "".
 func (a *Driver) Diff(id, parent string) (archive.Archive, error) {
+	// Make sure the hardlinks are copied up before taking the diff
+	if err := exec.Command("auplink", path.Join(a.rootPath(), "mnt", id), "flush").Run(); err != nil {
```

Comment:
Oh I see. We're covered there as well.
On Oct 12, 2015 12:35, "Tõnis Tiigi" <notifications@github.com> wrote:

> In daemon/graphdriver/aufs/aufs.go
> <https://github.com/docker/docker/pull/16774#discussion_r41780745>:
>
> > @@ -350,6 +350,10 @@ func (a *Driver) Put(id string) error {
> >  // Diff produces an archive of the changes between the specified
> >  // layer and its parent layer which may be "".
> >  func (a *Driver) Diff(id, parent string) (archive.Archive, error) {
> > +	// Make sure the hardlinks are copied up before taking the diff
> > +	if err := exec.Command("auplink", path.Join(a.rootPath(), "mnt", id), "flush").Run(); err != nil {
>
> I think its already called on Unmount,
> https://github.com/docker/docker/blob/master/daemon/graphdriver/aufs/mount.go#L14
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/16774/files#r41780745>.
>